server/sonar-server/src/main/java/org/sonar/server/es/EsClientProvider.java

```diff
@@ -43,25 +43,25 @@
 
   private EsClient cache;
 
-  public EsClient provide(Settings settings) {
+  public EsClient provide(Configuration config) {
     if (cache == null) {
       TransportClient nativeClient;
       org.elasticsearch.common.settings.Settings.Builder esSettings = org.elasticsearch.common.settings.Settings.builder();
 
       // mandatory property defined by bootstrap process
-      esSettings.put("cluster.name", settings.getString(ProcessProperties.CLUSTER_NAME));
+      esSettings.put("cluster.name", config.get(ProcessProperties.CLUSTER_NAME).get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)